content/en/blog/_posts/2022-08-03-kms-v2-alpha.md

```diff
@@ -0,0 +1,95 @@
+1. **Observability:** Without some kind of trace ID, it's has been difficult to correlate events found in the various logs across `kube-apiserver`, KMS, and KMS plugins.
+
+The KMS v2 enhancement attempts to address all of these shortcomings, though not all planned features are implemented in the initial alpha release. Here are the improvements that arrived in Kubernetes v1.25:
+
+1. Support for KMS plugins that use a key hierarchy to reduce network requests made to the remote vault. To learn more, check out the [design details for how a KMS plugin can leverage key hierarchy](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3299-kms-v2-improvements#key-hierachy).
+1. Extra metadata is now tracked to allow a KMS plugin to communicate what key it is currently using with the `kube-apiserver`, allowing for rotation without API server restart. Data stored in etcd follows a more standard proto format to allow external tools to observe its state. To learn more, check out the [details for metadata](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3299-kms-v2-improvements#metadata).
+1. A dedicated status API is used to communicate the health of the KMS plugin with the API server. To learn more, check out the [details for status API](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3299-kms-v2-improvements#status-api).
+1. To improve observability, a new `UID` field is included in `EncryptRequest` and `DecryptRequest` of the v2 API. The UID is generated for each envelope operation. To learn more, check out the [details for observability](https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3299-kms-v2-improvements#Observability).
+
+### Sequence Diagram
+
+#### Encrypt Request
+
+```mermaid
+sequenceDiagram
+    participant etcd
+    participant kubeapiserver
+    participant kmsplugin
+    participant externalkms
+    kubeapiserver->>kmsplugin: encrypt request
+    alt using key hierarchy
+        kmsplugin->>kmsplugin: encrypt DEK with local KEK
+        kmsplugin->>externalkms: encrypt local KEK with remote KEK
+        externalkms->>kmsplugin: encrypted local KEK
+        kmsplugin->>kmsplugin: cache encrypted local KEK
+        kmsplugin->>kubeapiserver: return encrypt response <br/> {"ciphertext": "<encrypted DEK>", key_id: "<remote KEK ID>", <br/> "annotations": {"kms.kubernetes.io/local-kek": "<encrypted local KEK>"}}
+    else not using key hierarchy
+        %% current behavior
+        kmsplugin->>externalkms: encrypt DEK with remote KEK
+        externalkms->>kmsplugin: encrypted DEK
+        kmsplugin->>kubeapiserver: return encrypt response <br/> {"ciphertext": "<encrypted DEK>", key_id: "<remote KEK ID>", "annotations": {}}
+    end
+    kubeapiserver->>etcd: store encrypt response and encrypted DEK
```

Comment:
These are not rendering. Could you render it yourself (ideally, in Kubernetes colors - see https://kubernetes.io/docs/concepts/services-networking/ingress/ for an example), export that as SVG, and then use the SVG image instead?